### ISSUE TITLE
Cannot unmarshal array into Go struct field PkiListRolesResponse.data.keys of type map[string]interface {}

### ISSUE THREAD

**Issue by TheLonelyGhost** [hashicorp/vault-client-go#175]
# Expected Behavior

Attempts to list PKI secrets engine roles using the `(*vault.Client).Secrets.PkiListRoles()` method succeed with the response object responding to `(resp).Data.Keys` with a signature of `[]string`, indicating the slice of role names.

# Current Behavior

Attempts to list PKI secrets engine roles using the `(*vault.Client).Secrets.PkiListRoles()` method fail due to the following error:

```
json: cannot unmarshal array into Go struct field PkiListRolesResponse.data.keys of type map[string]interface {}
```

# Failure Information

Please include the version of Vault binary and the version of `vault-client-go` you're using.

Vault server: 1.11.4+ent
Vault Client Go: 0.3.2

## Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Start a copy of Vault (i.e., `vault server -dev`)
2. Set environment variables `VAULT_ADDR` and `VAULT_TOKEN` accordingly
3. Enable a PKI secrets engine
4. Create a role on the enabled PKI mount
5. Run the code below

```go
// File: example.go
package main

import (
	"context"
	"fmt"

	"github.com/hashicorp/vault-client-go"
)

func main() {
	client, err := vault.New(vault.WithEnvironment())
	if err != nil {
		panic(err)
	}

	out, err := client.Secrets.PkiListRoles(context.Background())
	if err != nil {
		panic(err)
	}

	fmt.Printf("Resp: %v\n", out.Data)
}
```

# Additional Information

The method signature of `PkiListRoles()` seems like it should follow other, similar "list" type activities, such as `AwsListRoles()`. The latter has a signature for `(*vault.Response[map[string]interface{}]).Data.Keys` of `[]string` whereas the former has `map[string]interface{}`. This might be as minor as having typed `{}` instead of `[]` somewhere.

**Comment by averche** [comment]
Hi @TheLonelyGhost, thanks for bringing this to our attention!

The root cause of the issue is an incorrect schema type in the generated openapi.json. It should be quite easy to fix. I'm addressing it in https://github.com/hashicorp/vault/pull/20782

**Comment by averche** [comment]
This issue should be resolved in [v0.3.3](https://github.com/hashicorp/vault-client-go/releases/tag/v0.3.3).
